### ISSUE TITLE
Offline problems

### ISSUE THREAD

**Issue by multun** [prologin/site-issues#33]
How about enabling users to download pdf exercises ?

**Comment by tobast** [comment]
I thought the same when I saw the forum post. I think it might be a good idea :)

**Comment by zopieux** [comment]
Possible solutions.

### *Download as pdf* feature on website
Generate LaTeX from the problem data and render a PDF. Would also require the ability to zip entire sections (each year exercices) to be useful.

Ideal (discoverable, easy to use, nice rendering quality) but time consuming. Technical solution: buildbot job on `/problems/` update, that builds PDFs and per-year zips.

### Public mirror of `/problems/` on GitHub
Mirror the content or `/problems/`, without private data (tests & other site-only properties). GitHub handles the boring stuff such as creating tarballs for users to download.

Easy to do but needs work to improve discoverability on the website.

**Comment by tobast** [comment]
@Zopieux is the first feature /that/ time consuming? Is it possible to re-use the latex rendering used for convocations, etc.? (I have no idea whether this is plain django/jinja, or a distinct module.)

If it is possible, am I right by saying that it would mostly be a matter of creating a tex template for a problem and adding a ``Download'' button in the problems' views? (The ``download whole year/event'' feature would require a bit more work, probably.)

**Comment by Zopieux** [comment]
@tobast the time-consuming part is rendering proper LaTeX from our Markdown sources. Converters such as Pandoc *may* work out-of-the-box, but I'm pretty sure we will have issues with eg. embedded LaTeX stubs. Also, references to static files and more generally speaking, embedded HTML.

**Comment by tobast** [comment]
Right, I indeed forgot that part. It seems though that we only use (and probably will ever use) only a small fragment of Markdown (which is not well-defined anyway), so maybe it would be possible to cook a simple, but working, converter?

**Comment by multun** [comment]
How about keeping it simple and using existing tools ;-) ?

**Comment by tobast** [comment]
Depends on how fine they work in our usecase, I guess. Of course a good first try would be to use pandoc.

**Comment by audebert** [comment]
Please keep the discussion on IRC and use this issue as work tracking.
